LabRecorder start/stop scripts

rchamplin opened the pull request:
I made some changes and added a step to "tomcat/human_experiments/scripts/run_experiment".
I finished developing the scripting in "tomcat/human_experiments/scripts/start_lab_recorder".
I created a script "tomcat/human_experiments/scripts/stop_lab_recorder" and a python script "tomcat/human_experiments/scripts/stop_recording.py".
All have been tested in last dry run and worked correctly.

adarshp commented on human_experiments/scripts/run_experiment:
```diff
@@ -30,7 +30,8 @@ __help() {
   echo "[11]: Starts the Client Map"
   echo "[12]: Opens the Final Survey"
-  echo "[13]: Ends the experiment (cleanup and backup phase)"
+  echo "[13]: Stop the Lab Recorder recordings and Kill the Lab Recorder Instances on CAT"
```
Could you remove the unnecessary titlecasing here? That is, make the following changes:

- `Kill` -> `kill`
- `Instances` -> `instances`

Additionally, could you change `Lab Recorder` to `LabRecorder` to match the way it's written about in the LSL docs?

paulosoaresua replied:
Also, the cleanup phase was supposed to do more things like moving data from Cat to Tom for backup, and kill everything else that it's open not just the LabRecorder instances. That's why it was named more generically.

adarshp replied:
@paulosoaresua - the cleanup phase has not been removed, it has just been moved to step 14 (see line 34 of the version of this file on this branch)

adarshp commented on human_experiments/scripts/run_experiment:
```diff
@@ -70,7 +71,7 @@ fi
 source check_global_variables
 source check_data_root_directory
-source check_connections
+#source check_connections
```
Why is this line commented out?

adarshp commented on human_experiments/scripts/run_experiment:
```diff
@@ -181,7 +182,16 @@ fi
 if [[ $GOTO -le 13 ]]; then
   echo " "
-  text="[13] Type [${EMPH}quit${NC}] to end the experiment and close "
+  yes_no_question "[13] Do you want to stop ${EMPH}Lab Recorder recordings and Kill the Lab Recorder Instances on CAT${NC}?"
```
Same as the above comment (remove unnecessary titlecasing, change Lab Recorder to LabRecorder)

paulosoaresua replied:
I preferred not to have this as a yes/no question to avoid having us ending an experiment by mistake. Typing `quit` explicitly means we really want to end it. I'm okay with yes/no if everyone else agree with it.

adarshp commented on human_experiments/scripts/start_lab_recorder:
```diff
@@ -19,4 +19,62 @@ else
+echo " "
+ssh cat@10.192.234.250 ps -F -C LabRecorder 
+echo " "
+echo -e "${YELLOW}Verify that you have 3 LabRecorder Windows open on CAT and that they are all recording the correct streams!${NC}"
```
Remove unnecessary titlecasing (`Windows` -> `windows`)

adarshp commented on human_experiments/scripts/start_lab_recorder:
```diff
@@ -19,4 +19,62 @@ else
+ssh cat@10.192.234.250 ps -F -C LabRecorder 
+echo " "
+echo -e "${YELLOW}Verify that you have 3 LabRecorder Windows open on CAT and that they are all recording the correct streams!${NC}"
+echo -e "${EMPH}(The 3 Windows on CAT might be stacked on top of each other so, "
```
`Windows` -> `windows`

adarshp commented on human_experiments/scripts/stop_lab_recorder:
```diff
@@ -0,0 +1,29 @@
+ssh cat@10.192.234.250 pkill LabRecorder
+echo " "
+echo -e "${YELLOW}KILLED: any Instances of LabRecorder running on CAT!${NC}"
+echo -e "${EMPH}(Verify that no LabRecorder Windows are open on CAT!)${NC}"
```
`Windows` -> `windows`

adarshp commented on human_experiments/scripts/stop_lab_recorder:
```diff
@@ -0,0 +1,29 @@
+echo -e "${YELLOW}KILLED: any Instances of LabRecorder running on CAT!${NC}"
+echo -e "${EMPH}(Verify that no LabRecorder Windows are open on CAT!)${NC}"
+echo " "
+
```
Remove blank line

adarshp commented on human_experiments/scripts/stop_recording.py:
```diff
@@ -0,0 +1,21 @@
+import socket
+import subprocess, os
```
The `subprocess` and `os` modules are not being used, so you can remove this line.

adarshp commented on human_experiments/scripts/stop_recording.py:
```diff
@@ -0,0 +1,21 @@
+time.sleep(10.0)
+
+sys.exit("All LabRecorder Recordings Stopped (Lion, Tiger and Leopard)!")
+
```
Remove blank line

adarshp commented on human_experiments/scripts/stop_recording.py:
```diff
@@ -0,0 +1,21 @@
+
+time.sleep(10.0)
+
+sys.exit("All LabRecorder Recordings Stopped (Lion, Tiger and Leopard)!")
```
Remove unnecessary titlecasing (`Recordings Stopped` -> `recordings stopped`)

adarshp commented on human_experiments/scripts/stop_recording.py:
```diff
@@ -0,0 +1,21 @@
+import time
+import sys
+
+s1 = socket.create_connection(("localhost", 22345))
```
Can you change this to a for loop? Something like:

```python
for port in (22345, 22346, 22347):
    s = socket.create_connection(("localhost", port))
    s.sendall(b"stop\n")
    s.close()
```

adarshp commented on human_experiments/scripts/stop_lab_recorder:
```diff
@@ -0,0 +1,29 @@
+echo " "
+echo -e "${YELLOW}WARNING: This will stop all 3 Instances of LabRecorder on CAT from Recording!"
+echo -e "         (Lion, Tiger and Leopard)"
+echo -e "         And KILL any Instances of LabRecorder running on CAT!${NC}"
```
Remove unnecessary titlecasing (`Instances` -> `instances`)

adarshp commented on human_experiments/scripts/stop_lab_recorder:
```diff
@@ -0,0 +1,29 @@
+source configuration_helpers
+
+echo " "
+echo -e "${YELLOW}WARNING: This will stop all 3 Instances of LabRecorder on CAT from Recording!"
```
Instances -> instances
Recording -> recording

adarshp commented on human_experiments/scripts/stop_lab_recorder:
```diff
@@ -0,0 +1,29 @@
+#Kill any Instances of LabRecorder running on CAT.
+ssh cat@10.192.234.250 pkill LabRecorder
+echo " "
+echo -e "${YELLOW}KILLED: any Instances of LabRecorder running on CAT!${NC}"
```
Instances -> instances

adarshp commented on human_experiments/scripts/start_lab_recorder:
```diff
@@ -19,4 +19,62 @@ else
+#Will need to open 3 fresh instances of LabRecorder to gaurentee they using correct config file,
+#connecting to correct streams and using Experiment Dir and Data File for this Experiment.  
+echo " "
+echo -e "Opening 3 Instances of LabRecorder on CAT:"
```
Instances -> instances

adarshp commented on human_experiments/scripts/start_lab_recorder:
```diff
@@ -19,4 +19,62 @@ else
+#Kill any prior instances of LabRecorder on CAT so that fresh instances can be opened.
+ssh cat@10.192.234.250 pkill LabRecorder
+echo " "
+echo -e "${YELLOW}KILLED: any Prior Instances of LabRecorder running on CAT!${NC}"
```
Prior Instances -> prior instances

adarshp commented on human_experiments/scripts/start_lab_recorder:
```diff
@@ -19,4 +19,62 @@ else
+
+
+echo " "
+echo -e "Configuring LabRecorder's Config Files with Data Path/File:"
```
Config -> config
Files -> files
Data Path/File -> data path/file

adarshp commented on human_experiments/scripts/start_lab_recorder:
```diff
@@ -19,4 +19,62 @@ else
-# TODO
\ No newline at end of file
+echo " "
+echo -e "${YELLOW}WARNING: All Prior Instances of LabRecorder running on CAT will be KILLED!${NC}"
```
Remove unnecessary titlecasing

adarshp commented:
Hi @rchamplin , I've added a few comments on the code - could you please resolve them? Additionally, could you please update the title of the pull request to be more descriptive of the actual changes contained in it?

paulosoaresua reviewed (commented):
Added come comments.

rchamplin commented:
Pushed again to my branch dev_rchamplin.
5 files updated.

Fixed my "unneeded" title casing in all files per request.

end_experiment: line 99, for the name of app being closed on iMACs, change from "Chrome" to "Google Chrome". When Chrome is open on iMacs, the OS refers to the app as "Google Chrome" not just "Chrome". issuing a command in uber script to close "Chrome" on iMACs was causing the script to pause and iMACs opening a dialog window stating "select what app you want to close". Changing the app name to "Google Chrome" fixed this problem.

run_experiment: uncommented out line 74 "source check_connections". Not sure who commented this line out. However, it was determined that we want the connections to the iMACs to be checked at beginning of experiment.
Added a step 13 ( Stop LabRecorder recordings and kill the LabRecorder instances on CAT - "stop_lab_recorder") to the script and moved what was step 13 to step 14 (End the experiment and close all open applications and processes - "end_experiment").

start_lab_recorder: This script was never finished. It just had code to bring in the directory path command line argument for the experiment. I developed the process and finished script.
This script first modifies the "StorageLocation=" line in each unique LabRecorder config file (LabRecorder_Lion_22345.cfg, LabRecorder_Tiger_22346.cfg, LabRecorder_Leopard_22347.cfg) to have the correct experiment directory and XDF file name.
Second, close any open instances of LabRecorder running on CAT.
Third, open 3 instances of LabRecorder in background on CAT using the default console display for the GUI windows and using the unique config file for each instance (Lion port: 22345, Tiger port: 22346, Leopard port: 22347) that will setup that instance with the unique port, path, file name, and streams it should be using and, automatically put it in record mode after initializing.

stop_lab_recorder: This script had never been developed before, I just created it.
First, by calling a python script I created (stop_recording.py), it puts the 3 instances of LabRecorder running on CAT into "Stop recording mode".
Second, after waiting 10 seconds (delay coded into stop_recording.py), it kills the 3 instances of LabRecorder running on CAT.

stop_recording.py: Python script I created. Note: I changed the straight line coding that I had in the script to a for loop per requested.
First, it issues a "stop" (stop recording) API command to each instance of LabRecorder on the unique port each is listening on (Lion port: 22345, Tiger port: 22346, Leopard port: 22347).   
Second, it does a 10 second pause to give time for each instance to switch to "stop recording mode" and close its XDF files and handles before continuing.
